pinot-core/src/main/java/org/apache/pinot/core/operator/GroupByRecord.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator;
+
+import com.google.common.base.Joiner;
+import javax.annotation.Nonnull;
+import org.apache.pinot.common.utils.EqualityUtils;
+import org.apache.pinot.core.query.aggregation.function.AggregationFunction;
+
+
+/**
+ * The <code>GroupByRecord</code> class represents one record of a group by query result
+ */
+public class GroupByRecord {
+  private String[] _groupByKey;
+  private Object[] _aggregationResults;
+  // TODO: could just 1 array be used to store both keys and values?
+
+  private static String SEPARATOR = "\t";
```

Comment:
We should define this at a common place where both the split and join callers can use it.

Reply:
Done